test/utils/runners.go

```diff
@@ -1588,10 +1589,23 @@ func attachSecrets(template *v1.PodTemplateSpec, secretNames []string) {
 			Name:      name,
 			MountPath: fmt.Sprintf("/%v", name),
 		})
+		envFrom = append(envFrom, v1.EnvFromSource{
+			Prefix: name,
+			SecretRef: &v1.SecretEnvSource{
+				LocalObjectReference: v1.LocalObjectReference{
+					Name: name,
+				},
+			},
+		})
+		imagePullSecrets = append(imagePullSecrets, v1.LocalObjectReference{
+			Name: name,
+		})
 	}
 
+	template.Spec.ImagePullSecrets = imagePullSecrets
```

Comment:
I don't think you can put any random secret here. This will probably cause the image to fail to be pulled? Have you been able to run this test case to verify it still works?

Reply:
That is a nice point.
In addition, the secret of imagePullSecret is `type: kubernetes.io/dockerconfigjson`.
That is different format from generic secrets which are passed from this function caller sides.
I think we should skip implementing imagePullSecret here.